### PR TITLE
`dmesg` improvements

### PR DESCRIPTION
**Pull request by no92**
This PR ensures that logging to dmesg is pretty (by using priorities and stripping CSI sequences) and works as expected. This includes eudev's logging to `/dev/kmsg`, which uses prefixes as defined in RFC 5424.

Needs to be merged along with managarm/mlibc#1090.

### CODE REVIEW

File: kernel/thor/generic/hel.cpp [hunk 1]
```diff
@@ -125,10 +147,57 @@ HelError helLog(const char *string, size_t length) {
 		if(!readUserArray(string + offset, log.text, chunk))
 			return kHelErrFault;
 
-		auto p = infoLogger();
-		for(size_t i = 0; i < chunk; i++)
-			p << frg::char_fmt(log.text[i]);
-		p << frg::endlog;
+		switch(severity) {
+			case kHelLogSeverityEmergency: {
+				auto p = panicLogger();
```
Comment (qookei) [inline kernel/thor/generic/hel.cpp:152]: I don't think letting the user trigger a kernel panic is a good idea :^)

Comment (no92) [reply kernel/thor/generic/hel.cpp]: Fixed by rolling it into the case below.

File: kernel/thor/generic/ubsan.cpp [hunk 2]
```diff
@@ -73,6 +73,10 @@ namespace {
 	}
 }
 
+extern "C" void abort() {
```
Comment (Dennisbonke) [inline kernel/thor/generic/ubsan.cpp:76]: Why is this added? It seems unrelated.

**Review by Dennisbonke — COMMENTED** [review]
One comment.

File: kernel/thor/generic/ubsan.cpp (hunk 2 above)
Comment (no92) [reply kernel/thor/generic/ubsan.cpp]: Fixed.

**Review by Dennisbonke — APPROVED** [review]
LGTM
